lib/engine-native/src/artifact.rs

```diff
@@ -593,7 +602,117 @@ impl Artifact for NativeArtifact {
     }
 
     fn register_frame_info(&self) {
-        // Do nothing for now
+        let mut info = self.frame_info_registration.lock().unwrap();
+
+        if info.is_some() {
+            return;
+        }
+
+        // The function sizes might not be completely accurate.
+        // Because of that, we (reverse) order all the functions by their pointer.
+        // [f9, f7, f6, f8...] and calculate their potential function body size by
+        // getting the diff in pointers between functions.
```

Comment:
Oh, I think there was a misunderstanding on this.

The algorithm doesn't require functions to be sequentially ordered.
What we do, is:
1. we have the functions ordered by index first [f1, f2, f3, f4, f5]
2. we switch the order to the inverse memory order: [f3, f5, f4, f2, f1] (that means that the pointer of f3 is the highest, then f5, then f4 and so on...).
3. Because all the functions are in the same `__text` section (the executable section of the object file), we guess the function size by looking at the difference between the previous.

In this case, the f5 estimated function body size will be the (pointer of f3) - (pointer of f5)

Reply:
I've re-done the text trying to make it clear